### ISSUE TITLE
Bug in expression evaluation

### ISSUE THREAD

**Issue by deepakjois** [deepakjois/hs-logo#31]
```
pr 3 * (5 + 2) / 4
pr 3 * 7 / 4
pr 3 * 7 + 4
```

yields

```
21.0
21.0
25.0
```

**Comment by deepakjois** [comment]
Currently, there is good way to treat an expression containing one operators (and two operands) and an expression containing two operators (and three operators) the same. The precedence rules are a bit unclear when that happens. Do we do a function application first, or do we evaluate the full expression first before we do the function application.

The current workaround is to convert every expression into something that has two operators by using parentheses.

**Comment by deepakjois** [comment]
Fixed in fc3795e
